### ISSUE TITLE
dx bundle fail on windows

### ISSUE THREAD

**Issue by pojoin** [DioxusLabs/dioxus#3233]
**ERROR Failed to bundle project: BundlerError**

```
371.923s  INFO Copying app to output directory...
 375.582s ERROR Failed to bundle project: BundlerError(
    Error {
        context: "error running candle.exe",
        source: GenericError(
            "failed to run C:\\Users\\hcq\\AppData\\Local\\tauri\\WixTools314\\candle.exe",
        ),
    },
)
 375.594s ERROR err=BundleFailed(BundlerError(error running candle.exe
```

**Comment by photino** [comment]
The same error.

**Comment by dunkyl** [comment]
I got it sort of working, after following workarounds:

0. (seems like many have already done this) added `icons/icon.ico`

1. I added to Dioxus.toml:
```toml
 [bundle]
publisher = "my example publisher"
identifier = "com.example.app"
```
2. I renamed `target/debug/<APPNAME>/bundle/windows/<APPNAME>` to `<APPNAME>.exe`

3. I manually ran WiX:
```sh
~\AppData\Local\tauri\WixTools314\light.exe -ext WixUIExtension -loc .\target/dx/<APPNAME>/bundle/windows\wix\x64\locale.wxl .\target\dx\<APPNAME>\bundle\windows\wix\x64\main.wixobj
```
This successfully generated a working installer, however, app lacked permission to create its WebView2 cache in the default installed location (`C:\Program Files\<APPNAME>\`). Installing, to a different location, under my user folder, worked fine.

**Comment by dunkyl** [comment]
dx --version: `dioxus 0.6.0 (ccf500c)`

**Comment by liigo** [comment]
The same error:

```shell
  17.829s  INFO Bundling app...
  17.852s  INFO Copying asset (0/3): C:\Users\liigo\tmp\hidx06\myhotdog\assets\header.svg
  17.852s  INFO Copying app to output directory...
  18.820s ERROR Failed to bundle project: BundlerError(
    Error {
        context: "error running candle.exe",
        source: GenericError(
            "failed to run C:\\Users\\liigo\\AppData\\Local\\tauri\\WixTools314\\candle.exe",
        ),
    },
)
  18.823s ERROR err=BundleFailed(BundlerError(error running candle.exe

Caused by:
    `failed to run C:\Users\liigo\AppData\Local\tauri\WixTools314\candle.exe`))
```

But tauri build works well (both tauri and dioxus use *the same wix* as i known):

```shell
PS C:\Users\liigo\tmp\hi-tauri-dioxus> cargo tauri build
    Running beforeBuildCommand `dx build --release`
   0. 1s  INFO Building project...
   ......
   5.461s  INFO Build completed successfully! 🚀 path="C:\\Users\\liigo\\tmp\\hi-tauri-dioxus\\target\\dx\\hi-tauri-dioxus-ui\\release\\web\\public"
   Compiling hi-tauri-dioxus v0.1.0 (C:\Users\liigo\tmp\hi-tauri-dioxus\src-tauri)
    Finished `release` profile [optimized] target(s) in 1m 12s
    Built application at: E:\tmp\RUST_DIR\CARGO_TARGET_DIR\release\hi-tauri-dioxus.exe
    Info Target: x64
    Running candle for "main.wxs"
    Running light to produce E:\tmp\RUST_DIR\CARGO_TARGET_DIR\release\bundle\msi\hi-tauri-dioxus_0.1.0_x64_en-US.msi
    Finished 1 bundle at:
        E:\tmp\RUST_DIR\CARGO_TARGET_DIR\release\bundle\msi\hi-tauri-dioxus_0.1.0_x64_en-US.msi
```

**Comment by oletf** [comment]
during my exploration on #3688, I realized that the shortcut created by nsis (and probably msi as well) don't work anymore after #3482, 
since the shortcut created by the nsis bundler takes [`krate.bundled_app_name()`](https://github.com/DioxusLabs/dioxus/blob/68ca4c3289f3108ffdd6b8e0ad187071d10a110a/packages/cli/src/cli/bundle.rs#L211), to which it adds `.exe`.

the resulting link targets `appname.exe.exe`

**Comment by oletf** [comment]
alright !
 I found the culprit for the shortcut problem, it was on the [executable name here](https://github.com/DioxusLabs/dioxus/blob/68ca4c3289f3108ffdd6b8e0ad187071d10a110a/packages/cli/src/cli/bundle.rs#L157), #3482 helped a lot!

The shortcut, the uninstaller not deleting the executable, as well as WiX still not working [on my first reply here](https://github.com/DioxusLabs/dioxus/issues/3233#issuecomment-2634448102), all now work on my side !

1 line PR incoming
